lib/exprengine.cpp

```diff
@@ -971,6 +975,44 @@ static bool isNonOverlapping(ExprEngine::ValuePtr v1, ExprEngine::ValuePtr v2)
     return false;
 }
 
+static bool conditionAlwaysFalse(ExprEngine::ValuePtr condValue, ExprEngine::DataBase *dataBase)
+{
+    if (auto v = std::dynamic_pointer_cast<ExprEngine::IntRange>(condValue)) {
+        if (v->hasValue(0))
+            return true;
+        auto zero = std::make_shared<ExprEngine::IntRange>("0", 0, 0);
+        return ExprEngine::BinOpResult("==", v, zero).isTrue(dataBase);
```

Comment:
With this code you determine that the value _can be_ 0. So it is confusing that the function name is "alwaysFalse".

Reply:
This is one of the reasons I suggested that you restructure the code and say "canBeFalse" instead of "alwaysFalse".

Reply:
Yes. It takes few-hours debugging session to understood that.